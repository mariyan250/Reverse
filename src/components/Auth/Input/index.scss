@import '../../../index';

.auth-input {
  font-size: to-rem(16px);
}