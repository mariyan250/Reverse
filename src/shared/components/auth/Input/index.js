// React and style
import React from 'react';
import './index.scss';

// krastavici?
// auth input on both elements?
function Input(props) {
  return (
    <section className="auth-input">
      <label htmlFor={props.name}>{props.label}</label>
      <input {...props} className="auth-input p-1" type={props.type} name={props.name} />
    </section>
  );
}

export default Input;
