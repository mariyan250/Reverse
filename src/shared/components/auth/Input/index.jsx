// React and style
import React from 'react';
import styles from './index.module.scss';

function Input({ name, label, type, required, onChange }) {
  return (
    <section className={`${styles.input}`}>
      <label htmlFor={name}>{label}</label>
      <input className={`p-1`} type={type} name={name} required={required} onChange={onChange} />
    </section>
  );
}

export default Input;
