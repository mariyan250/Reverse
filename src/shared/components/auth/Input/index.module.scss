@import '../../../../index.scss';

.input {
  font-size: to-rem(16px);
}
