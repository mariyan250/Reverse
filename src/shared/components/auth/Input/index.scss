@import '../../../../index.scss';

// to variable please :) or to class, even better!
.auth-input {
  font-size: to-rem(16px);
}
