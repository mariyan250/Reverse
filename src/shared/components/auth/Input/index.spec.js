import React from 'react';
import { renderWithRouter } from 'tests/test-utils';
import Input from './index';

describe('Input', () => {
  let initialState;

  beforeAll(() => {
    initialState = {};
  });

  describe('Rendering', () => {
    it('should render the component with initial state', () => {
      const { container } = renderWithRouter(<Input />, initialState);

      expect(container.querySelector('.auth-input')).toMatchSnapshot();
    });
  });
});
