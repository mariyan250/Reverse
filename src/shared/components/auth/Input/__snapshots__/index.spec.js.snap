// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Rendering should render the component with initial state 1`] = `
<section
  class="auth-input"
>
  <label />
  <input
    class="p-1"
  />
</section>
`;
